src/utils.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(Debug)]
```

Comment:
I think it would be a good idea to rename this file to `timer.rs` or move it to e.g. `utils/timer.rs`. "utils" is a very general name and we might want to add more "helper types" like this in the future so it is better to draw the lines now I think 🙂

Also, just a hint, this module looks so unit-testable 🐻

Reply:
Sounds good, I pushed an update.